financial-templates-lib/clients/TokenBalanceClient.js

```diff
@@ -28,23 +28,23 @@ class TokenBalanceClient {
   }
 
   // Delete all data within the client.
-  clearState = async () => {
+  clearState = () => {
     this.tokenBalances = { collateralBalances: {}, syntheticBalances: {}, etherBalances: {} };
     this.accountMonitorList = [];
   };
 
-  getCollateralBalance = address => {
-    this._registerAddress(address);
+  getCollateralBalance = async address => {
```

Comment:
This seems to diverge a bit from how most of the other clients work. IIUC, most use the update method to cache on-chain state and then the `get` functions just return info from that cached data. I get why this one works a little differently -- because the addresses are not known until they're passed into the `get` methods.

From a UX perspective, this client now feels like a hybrid between updating immediately on call to `get` and updating during `update()`. Thoughts on just passing in a list of addresses on construction or in the `update()` method? Not super opinionated here -- just thought it might make sense to revisit the designs on this client now that it's being used differently.

Reply:
Ye, this is a super good point and I agree that how it was implemented was not ideal as it deviates from the other clients. It is not great for them to by async. 

To deal with this I've come up with a bit of a combination solution from your feedback. I don't want to add this to the constructor as this would require the monitor bot's `index.js` to iterate over these params and feed them in at construction time. Likewise, I don't want to feed this into the `update()` function as this will give it a different interface. 

Rather, I have added a `batchRegisterAddresses()` function that called by the constructor of the client implementor to specify addresses it wants to be populated into the client. In this implementation, the `BalanceMonitor.js` tells the client to watch these specific addresses and as such, they are ready to be queried on the first `update()` call.

In my mind this is a good solution because we keep the client implementation synchronous like the others and it does not place any burden on the root of the bots.

LMKWYT :)

Reply:
SGTM, I like this solution!